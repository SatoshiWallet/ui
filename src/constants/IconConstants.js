// @flow

// Simple
export const SIMPLE_ICONS = 'simpleIcons'
export const CHECK_CIRCLE = 'check'
export const ARROW_RIGHT = 'arrow-right'

// Entypo
export const ENTYPO = 'entypo'
export const THREE_DOT_MENU = 'dots-three-vertical'

// FontAwesome
export const FONT_AWESOME = 'fontAwesome'
export const ADDRESS_BOOK_O = 'address-book-o'
export const USER_O = 'user-o'
export const DELETE = 'trash-o'
export const RESYNC = 'refresh'
export const SPLIT = 'code-fork'
export const RENAME = 'edit'
export const GET_SEED = 'user-secret'
export const EXCHANGE_ICON = 'exchange'
export const QUESTION_ICON = 'question'
export const FA_MONEY_ICON = 'money'
export const FLAG = 'flag-o'

// Feather
export const FEATHER_ICONS = 'featherIcons'
export const SHOPPING_CART = 'shopping-cart'

// Ionic
export const ION_ICONS = 'ionIcons'
export const CLOSE_ICON = 'ios-close'
export const UNLOCKED_ICON = 'ios-unlock'
export const LOCKED_ICON = 'ios-lock'
export const EYE_ICON = 'eye'
export const MAIL = 'ios-mail'
export const ALERT_OUTLINE = 'ios-alert-outline'
export const KEY_ICON = 'ios-key'
export const WALLET_ICON = 'ios-wallet'
export const IOS_INFO_ICON = 'ios-information-circle-outline'
export const ANDROID_INFO_ICON = 'md-information-circle-outline'
export const WARNING = 'ios-warning'

// Material
export const MATERIAL_ICONS = 'material'
export const KEYBOARD_ARROW_DOWN = 'keyboard-arrow-down'
export const KEYBOARD_ARROW_UP = 'keyboard-arrow-up'
export const SWAP_VERT = 'swap-vert'
export const SWAP_HORIZ = 'swap-horiz'
export const EXCLAMATION = 'exclamation'

// MaterialCommunity
export const MATERIAL_COMMUNITY = 'materialCommunity'
export const ALERT = 'alert-circle-outline'
export const ARROW_DOWN_THICK = 'arrow-down-thick'
export const ARROW_DOWN_BOLD = 'arrow-down-bold'
export const ARROW_DOWN = 'arrow-down'
export const PICKAXE = 'pickaxe'
export const TELESCOPE = 'telescope'
export const SERVER = 'server'
