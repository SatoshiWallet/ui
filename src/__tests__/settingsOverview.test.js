/* globals jest describe it expect */
/* eslint-disable flowtype/require-valid-file-annotation */

import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import SettingsOverview from '../components/scenes/SettingsOverviewScene.js'
import { LOCKED_ICON, UNLOCKED_ICON } from '../constants/IconConstants.js'

describe('MyComponent', () => {
  it('should render UnLocked SettingsOverview', () => {
    const renderer = new ShallowRenderer()

    const props = {
      defaultFiat: '',
      autoLogoutTimeInSeconds: 0,
      username: '',
      account: { currencyConfig: {} },
      pinLoginEnabled: false,
      supportsTouchId: false,
      touchIdEnabled: false,
      lockButton: '',
      lockButtonIcon: UNLOCKED_ICON,
      isLocked: false,
      confirmPasswordError: '',
      sendLogsStatus: '',

      setAutoLogoutTimeInSeconds: jest.fn,
      confirmPassword: jest.fn,
      lockSettings: jest.fn,
      dispatchUpdateEnableTouchIdEnable: jest.fn,
      sendLogs: jest.fn,
      resetConfirmPasswordError: jest.fn,
      resetSendLogsStatus: jest.fn,
      onTogglePinLoginEnabled: jest.fn,
      showRestoreWalletsModal: jest.fn
    }
    const actual = renderer.render(<SettingsOverview {...props} />)

    expect(actual).toMatchSnapshot()
  })

  it('should render Locked SettingsOverview', () => {
    const renderer = new ShallowRenderer()

    const props = {
      defaultFiat: '',
      autoLogoutTimeInSeconds: 0,
      username: '',
      account: { currencyConfig: {} },
      pinLoginEnabled: false,
      supportsTouchId: false,
      touchIdEnabled: false,
      lockButton: '',
      lockButtonIcon: LOCKED_ICON,
      isLocked: true,
      confirmPasswordError: '',
      sendLogsStatus: '',

      setAutoLogoutTimeInSeconds: jest.fn,
      confirmPassword: jest.fn,
      lockSettings: jest.fn,
      dispatchUpdateEnableTouchIdEnable: jest.fn,
      sendLogs: jest.fn,
      resetConfirmPasswordError: jest.fn,
      resetSendLogsStatus: jest.fn,
      onTogglePinLoginEnabled: jest.fn,
      showRestoreWalletsModal: jest.fn
    }
    const actual = renderer.render(<SettingsOverview {...props} />)

    expect(actual).toMatchSnapshot()
  })
})
