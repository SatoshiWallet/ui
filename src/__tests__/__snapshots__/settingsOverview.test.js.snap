// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render Locked SettingsOverview 1`] = `
<SceneWrapper
  background="body"
  hasTabs={false}
>
  <ScrollViewMock>
    <SettingsHeaderRow
      icon={
        <Icon
          allowFontScaling={false}
          color="#FFFFFF"
          name="user-o"
          size={31}
        />
      }
      text="Account: "
    />
    <SettingsRow
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#4A5157"
          name="ios-lock"
          size={31}
        />
      }
    />
    <SettingsRow
      disabled={true}
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Change Password"
    />
    <SettingsRow
      disabled={true}
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Change PIN"
    />
    <SettingsRow
      disabled={true}
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Setup 2 Factor"
    />
    <SettingsRow
      disabled={true}
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Set Up Password Recovery"
    />
    <SettingsHeaderRow
      icon={
        <Icon
          allowFontScaling={false}
          color="#FFFFFF"
          name="ios-options"
          size={31}
        />
      }
      text="Options"
    />
    <SettingsRow
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Exchange Settings"
    />
    <SettingsRow
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Spending Limits"
    />
    <SettingsLabelRow
      onPress={[Function]}
      right="Disabled"
      text="Auto Log Off After"
    />
    <SettingsLabelRow
      right=""
      text="Default Currency"
    />
    <SettingsSwitchRow
      onPress={[Function]}
      text="PIN Re-login"
      value={false}
    />
    <SettingsSwitchRow
      onPress={[Function]}
      text="Developer Mode"
    />
    <SettingsRow
      onPress={[Function]}
      text="Restore Wallets"
    />
    <SettingsRow
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Terms of Service"
    />
    <View
      style={
        Object {
          "marginBottom": 90,
          "padding": 32,
        }
      }
    >
      <PrimaryButton
        onPress={[Function]}
      >
        <Text>
          Send Logs
        </Text>
      </PrimaryButton>
    </View>
  </ScrollViewMock>
</SceneWrapper>
`;

exports[`MyComponent should render UnLocked SettingsOverview 1`] = `
<SceneWrapper
  background="body"
  hasTabs={false}
>
  <ScrollViewMock>
    <SettingsHeaderRow
      icon={
        <Icon
          allowFontScaling={false}
          color="#FFFFFF"
          name="user-o"
          size={31}
        />
      }
      text="Account: "
    />
    <SettingsRow
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#4A5157"
          name="ios-unlock"
          size={31}
        />
      }
    />
    <SettingsRow
      disabled={false}
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Change Password"
    />
    <SettingsRow
      disabled={false}
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Change PIN"
    />
    <SettingsRow
      disabled={false}
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Setup 2 Factor"
    />
    <SettingsRow
      disabled={false}
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Set Up Password Recovery"
    />
    <SettingsHeaderRow
      icon={
        <Icon
          allowFontScaling={false}
          color="#FFFFFF"
          name="ios-options"
          size={31}
        />
      }
      text="Options"
    />
    <SettingsRow
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Exchange Settings"
    />
    <SettingsRow
      onPress={[Function]}
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Spending Limits"
    />
    <SettingsLabelRow
      onPress={[Function]}
      right="Disabled"
      text="Auto Log Off After"
    />
    <SettingsLabelRow
      right=""
      text="Default Currency"
    />
    <SettingsSwitchRow
      onPress={[Function]}
      text="PIN Re-login"
      value={false}
    />
    <SettingsSwitchRow
      onPress={[Function]}
      text="Developer Mode"
    />
    <SettingsRow
      onPress={[Function]}
      text="Restore Wallets"
    />
    <SettingsRow
      right={
        <Icon
          allowFontScaling={false}
          color="#87939E"
          name="right"
          size={22}
        />
      }
      text="Terms of Service"
    />
    <View
      style={
        Object {
          "marginBottom": 90,
          "padding": 32,
        }
      }
    >
      <PrimaryButton
        onPress={[Function]}
      >
        <Text>
          Send Logs
        </Text>
      </PrimaryButton>
    </View>
  </ScrollViewMock>
</SceneWrapper>
`;
